src/coreclr/src/tools/aot/ILCompiler.Reflection.ReadyToRun/ReadyToRunMethod.cs

```diff
@@ -100,14 +100,31 @@ public class RuntimeFunction
         /// </summary>
         public int StartAddress { get; }
 
+        /// <summary>
+        /// The relative virtual address to the end of the code block
+        /// </summary>
+        public int EndAddress { get;  }
+
         /// <summary>
         /// The size of the code block in bytes
         /// </summary>
         /// /// <remarks>
         /// The EndAddress field in the runtime functions section is conditional on machine type
         /// Size is -1 for images without the EndAddress field
         /// </remarks>
-        public int Size { get; }
+        public int Size
+        {
+            get
+            {
+                if (_sizeCache < 0)
+                {
+                    _sizeCache = GetSize();
+                }
+                return _sizeCache;
+            }
+        }
+
+        private int _sizeCache;
```

Comment:
Don't we want to initialize it to a negative value like `-1`?  Also, have you considered dropping the `Cache` suffix, i.e. `_sizeCache` --> `_size` and the same for GC info?

Reply:
Or did you mean the condition above to read `<= 0`?

Reply:
You're right, sure, will do.